lib/polyfills/fetch.js

```diff
@@ -36,22 +36,62 @@ if (typeof global.fetch === 'undefined' || !global.fetch.toString().match(/\[nat
   require('whatwg-fetch');
 
   // Export the polyfilled items that were housed in the global.self namespace.
-  module.exports = {
-    fetch: global.self.fetch,
-    Headers: global.self.Headers,
-    Request: global.self.Request,
-    Response: global.self.Response,
-  };
+  // fetch itself should be the default export, but we need to also make sure
+  // that we export the other items for destructuring in case they're needed.
+  Object.defineProperties(global.self.fetch, {
+    fetch: {
+      enumerable: true,
+      writable: false,
+      value: global.self.fetch,
+    },
+    Headers: {
+      enumerable: true,
+      writable: false,
+      value: global.self.Headers,
+    },
+    Request: {
+      enumerable: true,
+      writable: false,
+      value: global.self.Request,
+    },
+    Response: {
+      enumerable: true,
+      writable: false,
+      value: global.self.Response,
+    }
+  });
+
+  module.exports = global.self.fetch;
 
   // Restore the original value of 'self' to the version we cached at the top.
   global.self = self;
 }
 else {
   // fetch is natively supported, export it and its support constructors.
-  module.exports = {
-    fetch: global.fetch,
-    Headers: global.Headers,
-    Request: global.Request,
-    Response: global.Response,
-  };
+  // fetch itself should be the default export, but we need to also make sure
```

Comment:
We're modifying the native fetch object? That doesn't seem safe.

Reply:
Yes, we are modifying the window.fetch function, but for the properties we're adding, it should be fine, as that should be treated as a function by any regular consumer, and is only being treated as an object for the purposes of destructuring the module exports. The other option would be to make 4 individual modules in that directory, one to export each of fetch/Headers/Request/Response as the default export of itself, and then we'd have to polyfill those individually as needed.